src/main/scala/beam/agentsim/agents/ridehail/DefaultRideHailDepotParkingManager.scala

```diff
@@ -0,0 +1,729 @@
+  *
+  * @param parkingFilePath
+  * @param valueOfTime
+  * @param tazTreeMap
+  * @param random
+  * @param boundingBox
+  * @param distFunction
+  * @param parkingStallCountScalingFactor
+  * @param beamServices
+  */
+class DefaultRideHailDepotParkingManager[GEO: GeoLevel](
+  parkingFilePath: String,
+  valueOfTime: Double,
+  geoQuadTree: QuadTree[GEO],
+  idToGeoMapping: scala.collection.Map[Id[GEO], GEO],
+  geoToTAZ: GEO => TAZ,
+  random: Random,
+  boundingBox: Envelope,
+  distFunction: (Location, Location) => Double,
+  parkingStallCountScalingFactor: Double = 1.0,
+  beamServices: BeamServices,
+  skims: Skims,
+  outputDirectory: OutputDirectoryHierarchy
+) extends RideHailDepotParkingManager[GEO]
+    with LazyLogging {
+
+  // load parking from a parking file, or generate it using the geo beam input
+  val (
+    rideHailParkingZones: Array[ParkingZone[GEO]],
+    rideHailParkingSearchTree: ParkingZoneSearch.ZoneSearchTree[GEO]
+  ) = if (parkingFilePath.isEmpty) {
+    logger.info(s"no parking file found. generating ubiquitous ride hail parking")
+    ParkingZoneFileUtils
+      .generateDefaultParkingFromGeoObjects(
+        geoQuadTree.values().asScala,
+        random,
+        Seq(ParkingType.Workplace)
+      )
+  } else {
+    Try {
+      ParkingZoneFileUtils.fromFile[GEO](parkingFilePath, random, parkingStallCountScalingFactor)
+    } match {
+      case Success((stalls, tree)) =>
+        logger.info(s"generating ride hail parking from file $parkingFilePath")
+        (stalls, tree)
+      case Failure(e) =>
+        logger.warn(s"unable to read contents of provided parking file $parkingFilePath, got ${e.getMessage}.")
+        logger.info(s"generating ubiquitous ride hail parking")
+        ParkingZoneFileUtils
+          .generateDefaultParkingFromGeoObjects(
+            geoQuadTree.values().asScala,
+            random,
+            Seq(ParkingType.Workplace)
+          )
+    }
+  }
+
+  private val chargingPlugTypesSortedByPower = rideHailParkingZones
+    .flatMap(_.chargingPointType)
+    .distinct
+    .sortBy(-ChargingPointType.getChargingPointInstalledPowerInKw(_))
+
+  // track the usage of the RHM agency parking
+  var totalStallsInUse: Long = 0
+  var totalStallsAvailable: Long = 0
+
+  val parkingZoneSearchConfiguration: ParkingZoneSearchConfiguration =
+    ParkingZoneSearchConfiguration(
+      DefaultRideHailDepotParkingManager.SearchStartRadius,
+      DefaultRideHailDepotParkingManager.SearchMaxRadius,
+      boundingBox,
+      distFunction
+    )
+
+  ParkingZoneFileUtils.toCsv(
+    rideHailParkingZones,
+    outputDirectory.getOutputFilename(DefaultRideHailDepotParkingManager.outputRidehailParkingFileName)
+  )
+
+  val mnlMultiplierParameters: Map[ParkingMNL.Parameters, UtilityFunctionOperation] = Map(
```

Comment:
This manager is used for CAVs  (_robots_), so we don't need to pass `mnlMultiplierParameters`, it is created here